File: otherLangs/haskell/01.hs
import Control.Monad ()
import Data.List.Split ( splitOn )

main :: IO ()
main = do
  contents <- readFile "inputs/01.txt"
  let input = readInts (splitOn "\n" contents)
  print $ product $ head $ filter ((2020==) . sum) $ combinations 2 input
  print $ product $ head $ filter ((2020==) . sum) $ combinations 3 input

readInts :: [String] -> [Int]
readInts = map read

combinations :: Int -> [Int] -> [[Int]]
combinations 1 ns = map (:[]) ns
combinations k ns = concat (zipWith
       (\t i -> map (t:) (combinations (k - 1) (drop i ns)))
    ns [1..])

File: 2015/01.js
import { readFile, sum } from '../tools.ts';

console.log(sum(readFile('./inputs/01.txt').split('').map(c => ({'(':1,')':-1}[c]))))

const input = readFile('./inputs/01.txt')

let depth = 0;
for(let i = 0; i < input.length; i++){
  const c = input[i];
  if(c === '(') depth++;
  else {
    if(depth === 0) {
      console.log(i + 1)
      break;
    }
    depth--;
  }
}

